dirty_cat/_fuzzy_join.py

```diff
@@ -7,31 +7,155 @@
+
+
+def _numeric_encoding(main, main_cols, aux, aux_cols):
+    """Encoding numerical columns.
+
+    Parameters
+    ----------
+    main : pd.DataFrame
+        A table with numerical columns.
+    main_cols : str or list
+        The columns of the main table.
+    aux : pd.DataFrame
+        Another table with numerical columns.
+    aux_cols : str or list
+        The columns of the aux table.
+
+    Returns
+    -------
+    main_array : array-like
+        An array of the encoded columns of the main table.
+    aux_array : array-like
+        An array of the encoded columns of the aux table.
+    """
+    aux_array = aux[aux_cols].to_numpy()
+    main_array = main[main_cols].to_numpy()
+    # Re-weighting to avoid measure specificity
+    scaler = StandardScaler()
+    scaler.fit(np.concatenate((aux_array, main_array)))
+    aux_array = scaler.transform(aux_array)
+    main_array = scaler.transform(main_array)
+    return main_array, aux_array
+
+
+def _string_encoding(main, main_cols, aux, aux_cols, encoder, analyzer, ngram_range):
+    """Encoding string columns.
+
+    Parameters
+    ----------
+    main : pd.DataFrame
+        A table with string columns.
+    main_cols : str or list
+        The columns of the main table.
+    aux : pd.DataFrame
+        Another table with string columns.
+    aux_cols : str or list
+        The columns of the aux table.
+    encoder: Union[Literal["hashing"], _VectorizerMixin]
+        Encoder parameter for the Vectorizer.
+        See fuzzy_join's docstring for more information.
+    analyzer : {"word", "char", "char_wb"}
+        Analyzer parameter for the HashingVectorizer passed to
+        the encoder and used for the string similarities.
+        See fuzzy_join's docstring for more information.
+    ngram_range : int 2-tuple, optional, default=(2, 4)
+        The lower and upper boundary of the range of n-values for different
+        n-grams used in the string similarity.
+        See fuzzy_join's docstring for more information.
+
+    Returns
+    -------
+    main_array : array-like
+        An array of the encoded columns of the main table.
+    aux_array : array-like
+        An array of the encoded columns of the aux table.
+    """
+    # Make sure that the column types are string and categorical:
+    main_cols_clean = main[main_cols].astype(str)
+    aux_cols_clean = aux[aux_cols].astype(str)
+
+    if isinstance(main_cols, list) and isinstance(aux_cols, list):
+        main_cols_clean = main_cols_clean[main_cols].apply(
+            lambda row: "  ".join(row.values.astype(str)), axis=1
+        )
+        aux_cols_clean = aux_cols_clean[aux_cols].apply(
+            lambda row: "  ".join(row.values.astype(str)), axis=1
+        )
+    all_cats = pd.concat([main_cols_clean, aux_cols_clean], axis=0).unique()
+
+    if encoder is None:
+        enc = HashingVectorizer(analyzer=analyzer, ngram_range=ngram_range)
+    else:
+        enc = encoder
```

Comment:
This could be simplified:
```suggestion
    if encoder is None:
        encoder = HashingVectorizer(analyzer=analyzer, ngram_range=ngram_range)
```
And then replace `enc` with `encoder` below.
I think we could also replace `enc_cv` by `encoder` as well.